docs/pages/kubernetes-access/controls.mdx

```diff
@@ -303,20 +239,295 @@ roleRef:
+Teleport Kubernetes Service intercepts the request and inspects the user's
+authorization. The Kubernetes Service denies the request if the user is not
+authorized to view a particular resource. If the user *is* authorized to perform
+their request, the Kubernetes Service modifies the request and forwards it to
+the appropriate API server.
+
+### Authorizing user requests
+
+When the Teleport Kubernetes Service receives a request, it evaluates two fields
+within the user's roles. If these fields do not allow the user to perform the
+request, the Kubernetes Service returns an error to the user:
+
+#### `kubernetes_labels`
+
+The Teleport Kubernetes Service will only allow a user access to a pod if the
+cluster where the pod is running has a label that matches a user's
+`kubernetes_labels` configuration.
+
+#### `kubernetes_resources`
+
+Some resource URIs within the Kubernetes API server include the names of
+specific resources.
+
+For example, if a user runs `kubectl exec` to execute a command against the
+`webapp` pod in the `development` namespace, `kubectl` sends a request to the
+target cluster's API server at the following path:
+
+```
+"/api/v1/namespaces/development/pods/webapp/exec"
+```
+
+If a Kubernetes pod is available within the URL path of the request to a
+Kubernetes API server, the Teleport Kubernetes Service will check whether the
+user is authorized to access that pod.
+
+In the example above, the Kubernetes Service checks if the user is authorized to
+access the `webapp` pod in the `development` namespace and, if not, denies the
+request.
+
+### Forwarding user requests
+
+Once the Teleport Kubernetes Service has authorized the user to perform a
+request against a Kubernetes cluster and (if applicable) a particular resource,
+it assembles a request to the upstream API server. It adds impersonation headers
+to the request based on the `kubernetes_groups` and `kubernetes_users` fields in
+the user's role (see the [discussion of these
+fields](#kubernetes_groups_and_kubernetes_users) earlier).
+
+Because the Teleport Kubernetes Service accesses the upstream API server via the
+RBAC principals listed in the Teleport user's `kubernetes_groups` and
+`kubernetes_users` fields, the principals you specify in these fields must have
+access to the resources listed in the user's `kubernetes_resources` field.
+Otherwise, the Kubernetes Service will forward a request to the upstream API
+server with improper authorization, and the API server will deny the request.
+
+### Multiple roles
+
+#### How the Kubernetes Service evaluates multiple roles
+
+Before evaluating a user's request to a Kubernetes API server, the Teleport
+Kubernetes Service checks each of the user's roles. If one role's
+`spec.allow.kubernetes_labels` or `spec.allow.kubernetes_resources` conditions
+do not match the user's request, the Kubernetes Service checks the next role,
+and so on. 
+
+If the Kubernetes Service finds a role with a `spec.allow` condition that
```

Comment:
Maybe it's worth mentioning that by using `kubernetes_labels` and `kubernetes_resources` you can leverage different access levels for pods by using different `kubernetes_groups` and `kubernetes_users`. We collect the allowed `kubernetes_users` and `kubernetes_groups` using the same logic we use for allowing access to a pod/cluster.